addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -98,23 +110,63 @@ public static boolean contains(String tag) {
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
+        SiteMap tree = Model.getSingleton().getSession().getSiteTree();
         List<Element> sourceElements = source.getAllElements();
         sourceElements.stream()
                 .filter(element -> SupportedElements.contains(element.getName()))
                 .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
                 .forEach(
-                        element ->
-                                newAlert()
-                                        .setRisk(Alert.RISK_MEDIUM)
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            String integrityHash = calculateIntegrityHash(msg, element, tree);
+                            String otherInfo =
+                                    integrityHash.isEmpty()
+                                            ? ""
+                                            : Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherInfo", integrityHash);
```

Comment:
I noticed an error in this while trying to sort out the UnitTests. The "otherinfo" needs to match the key name in messages.properties.

`otherinfo` not `otherInfo` (lower i) .... it was likely a mistake in my original example/comment or whatever.

Reply:
fixed it:)